File: server/model/vehicle.js
const mongoose = require('mongoose');

var VehicleSchema = new mongoose.Schema({
    model : {
        type : String,
        required: true
    },
    type : {
        type : String,
        required: true
    },
    capacity : {
        type : Number,
        required: true
    },
    regNo : {
        type: String,
        required: true,
        unique: true
    },
    ownerID : {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Owner',
        required: true
    },
    driverID : {
        type: mongoose.Schema.Types.ObjectId,
        ref: 'Driver'
    },
    location: {
        type : Object,
        latitude: {
            type: Number,
            required: true
        },
        longitude: {
            type: Number,
            required: true
        }
    }
})

module.exports = mongoose.model('vehicle', VehicleSchema);

File: TakeMEOwnerApp/app/src/main/java/com/example/TakeMEOwnerApp/Vehicle.java
package com.example.TakeMEOwnerApp;

public class Vehicle {
    String vehicle_id, model;
    int capacity;
    String owner_id;
    int reg_no;
    String driver_id;

    public Vehicle(String vehicle_id, String model, int reg_no) {
        this.vehicle_id = vehicle_id;
        this.model = model;
        this.reg_no = reg_no;
    }

    public Vehicle()
    {

    }
}
